rewrite-maven/src/main/java/org/openrewrite/maven/search/FindDependency.java

```diff
@@ -73,11 +100,31 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
         return new MavenIsoVisitor<ExecutionContext>() {
             @Override
             public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
-                if (isDependencyTag(groupId, artifactId)) {
+                if (isDependencyTag(groupId, artifactId) &&
+                    versionIsValid(version, versionPattern, () -> findDependency(tag))) {
                     return SearchResult.found(tag);
                 }
                 return super.visitTag(tag, ctx);
             }
         };
     }
+
+    private static boolean versionIsValid(@Nullable String desiredVersion, @Nullable String versionPattern,
+                                          Supplier<ResolvedDependency> resolvedDependencySupplier) {
+        if (desiredVersion == null) {
+            return true;
+        }
+        ResolvedDependency resolvedDependency = resolvedDependencySupplier.get();
+        if (resolvedDependency == null) {
+            // shouldn't happen, but if it does, fail the condition
+            return false;
+        }
+        String actualVersion = resolvedDependency.getVersion();
+        Validated<VersionComparator> validate = Semver.validate(desiredVersion, versionPattern);
+        if (validate.isInvalid()) {
+            return false;
+        }
+        assert(validate.getValue() != null);
+        return validate.getValue().isValid(actualVersion, actualVersion);
```

Comment:
Perhaps it's just my unfamiliarity, but calling `.isValid(actualVersion, actualVersion)` with the actual version twice seemed odd to me at first, although it might be ok from seeing how `Semver.validate` returns one of many ranges. Asking here just to be sure.

Reply:
The first arg of `VersionComparator::isValid` is for the current version, which is relevant for the `LatestPatch` implementation. In this case, the version is not changing, so the `actualVersion` is an appropriate value for both the current version and the version-under-test.

...but yes, it's a surprising method signature. I made another change recently in another class which had a long-standing (small) bug with its use of the same method.

Todo edit with link

Reply:
Thanks! I'll have a look there as well; Then we're good to merge if you think so too.